core/src/read/read_stream.rs

```diff
@@ -91,27 +106,32 @@ impl<D: Decoder> ReadDiskStream<D> {
 
         let file: PathBuf = file.into();
 
-        match ReadServer::new(
-            file,
-            start_frame,
-            stream_opts.num_cache_blocks + stream_opts.num_look_ahead_blocks,
-            stream_opts.block_size,
+        match ReadServer::spawn(
+            ReadServerOptions {
+                file,
+                start_frame,
+                num_prefetch_blocks: stream_opts.num_cache_blocks
```

Comment:
`stream_opts` should be destructured instead of accessing all fields individually. This ensures that you don't miss any field. Also applies to other parts of the code.

Should be improved by a follow-up PR.

Reply:
What do you mean by "destructured"?

Reply:
Ah, I know what you mean now.

Reply:
Alright, I did it.